perun-core/src/test/java/cz/metacentrum/perun/core/entry/ResourcesManagerEntryIntegrationTest.java

```diff
@@ -2696,7 +2696,33 @@ public void autoAssignSubgroupInactive() throws Exception {
 		List<AssignedGroup> assignedGroups = sess.getPerun().getResourcesManager().getGroupAssignments(sess, inactiveResource, List.of());
 
 		AssignedGroup assignedGroup = new AssignedGroup(new EnrichedGroup(group, List.of()), GroupResourceStatus.INACTIVE, null, null, true);
-		AssignedGroup assignedSubgroup = new AssignedGroup(new EnrichedGroup(subGroup, List.of()), GroupResourceStatus.INACTIVE, group.getId(), null, true);
+
+		// automatically assigned subgroups should always be ACTIVE
+		AssignedGroup assignedSubgroup = new AssignedGroup(new EnrichedGroup(subGroup, List.of()), GroupResourceStatus.ACTIVE, group.getId(), null, true);
+
+		assertThat(assignedGroups).containsExactlyInAnyOrder(assignedGroup, assignedSubgroup);
+	}
+
+	@Test
+	public void assignGroupsToResourceWithInactiveSourceGroupWithSubgroup() throws Exception {
+		System.out.println(CLASS_NAME + "autoAssignSubgroupsWithInactiveSourceGroup");
```

Comment:
Rename the print to match test name here too

Reply:
While we are at it, I would like if we named tests according cases they are testing. The code is more readable, if we know what it should do thanks to the test name. I suggest something like _assignInactiveGroupToResourceActivatesItsSubgroups_.